### PR TITLE
[IOTDB-279] Merge TsDigest into Statistics

### PR DESCRIPTION
**Pull request by jack870131**
Corresponding issue: https://issues.apache.org/jira/browse/IOTDB-279

Merge the function of TsDigest into Statistics class, which TsDisgest is the ByteBuffer format of Statistics.

### CODE REVIEW

**Comment by jack870131** [comment]
The message from CI:
```
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.016 s <<< FAILURE! - in org.apache.iotdb.db.writelog.WriteLogNodeManagerTest
[ERROR] testAutoSync(org.apache.iotdb.db.writelog.WriteLogNodeManagerTest)  Time elapsed: 0.016 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class sun.security.provider.SecureRandom$SeederHolder
	at org.apache.iotdb.db.writelog.WriteLogNodeManagerTest.testAutoSync(WriteLogNodeManagerTest.java:80)
```

Not sure if it is the error from my code.

**Comment by qiaojialin** [comment]
> The message from CI:
> 
> ```
> [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.016 s <<< FAILURE! - in org.apache.iotdb.db.writelog.WriteLogNodeManagerTest
> [ERROR] testAutoSync(org.apache.iotdb.db.writelog.WriteLogNodeManagerTest)  Time elapsed: 0.016 s  <<< ERROR!
> java.lang.NoClassDefFoundError: Could not initialize class sun.security.provider.SecureRandom$SeederHolder
> 	at org.apache.iotdb.db.writelog.WriteLogNodeManagerTest.testAutoSync(WriteLogNodeManagerTest.java:80)
> ```
> 
> Not sure if it is the error from my code.

I think this is a bug of our windows CI...

**Comment by jack870131** [comment]
@qiaojialin Received. I wanna make sure the merge process of IoTDB is. I mean, what kind of situation I could merge the pull request by myself as a committer? Or I need to wait more people to review my pr and let others merge it into Master branch?

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java [hunk 1]
```diff
@@ -299,4 +312,272 @@ public int serialize(OutputStream outputStream) throws IOException {
...
+    Statistics digest = getStatsByType(dataType);
+    int size = ReadWriteIOUtils.readInt(inputStream);
+    digest.validSizeOfArray = size;
+    digest.digestSerializedSize = Integer.BYTES;
+    if (size > 0) {
+      digest.statistics = new ByteBuffer[StatisticType.getTotalTypeNum()];
+      ByteBuffer value;
+      // check if it's an old version of TsFile
+      String key = "";
+      if (TSFileDescriptor.getInstance().getConfig().getEndian().equals("LITTLE_ENDIAN")) {
```
[the remark this answers is not in the document]
Comment (jack870131) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java]: Hi, thanks for your reply. I have corrected it.

**Pull request merged**